### PR TITLE
Handled exception from vault? method

### PR DESCRIPTION
**Pull request by sanga1794**
Signed-off-by: sanga17 <sausekar@msystechnologies.com>

## Description
Fixed 'vault?' method should throw meaningful exception when no data bag exists
## Related Issue
Fixed: https://github.com/chef/chef-vault/issues/278

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/chef-vault/item.rb [hunk 1]
```diff
@@ -340,7 +340,19 @@ def self.vault?(vault, name)
...
+      begin
+        dbi = Chef::DataBagItem.load(vault, name)
+      rescue Net::HTTPServerException => http_error
+        if http_error.response.code == "404"
+          raise ChefVault::Exceptions::ItemNotFound,
+            "#{vault}/#{name} could not be found"
+        else
+          raise http_error
+        end
+      rescue Chef::Exceptions::ValidationFailed
```
Comment (Vasu1105) [inline lib/chef-vault/item.rb:352]: @sanga1794 we are raising Item not found exception in case of ValidationFailed ?

Comment (sanga1794) [reply lib/chef-vault/item.rb:352]: just took the reference from method self.load https://github.com/chef/chef-vault/blob/6428e999cd724af42aa1da18060a2760b62393bf/lib/chef-vault/item.rb#L289

Comment (tas50) [inline lib/chef-vault/item.rb:352]: Does validation failing always indicate that the data bag isn't there? Would't that be the 404 you're rescuing above?

Comment (sanga1794) [reply lib/chef-vault/item.rb:352]: yes @tas50, in my case only rescuing 404 will be fine, will remove the remaining, thanks

File: lib/chef-vault/item.rb [hunk 2]
```diff
@@ -340,7 +340,16 @@ def self.vault?(vault, name)
     def self.data_bag_item_type(vault, name)
       # adapted from https://github.com/opscode-cookbooks/chef-vault/blob/v1.3.0/libraries/chef_vault_item.rb
       # and https://github.com/sensu/sensu-chef/blob/2.9.0/libraries/sensu_helpers.rb
-      dbi = Chef::DataBagItem.load(vault, name)
+      begin
+        dbi = Chef::DataBagItem.load(vault, name)
+      rescue Net::HTTPServerException => http_error
+        if http_error.response.code == "404"
+          raise ChefVault::Exceptions::ItemNotFound,
+            "#{vault}/#{name} could not be found"
```
Comment (kagarmoe) [inline lib/chef-vault/item.rb:348]: Consider ` "#{vault}/#{name} not found"`, unless we habitually use the polite subjunctive `could not be` in our output.

Comment (sanga1794) [reply lib/chef-vault/item.rb:348]: Thanks @kagarmoe, will update it.

**Comment by sanga1794** [comment]
> pry stack explorer needs to be pinned so these tests will pass. Here's how we're doing it in ohai:
> 
> https://github.com/chef/ohai/blob/master/Gemfile#L25



> pry stack explorer needs to be pinned so these tests will pass. Here's how we're doing it in ohai:
> 
> https://github.com/chef/ohai/blob/master/Gemfile#L25

Thanks @tas50, I'll update it

**Comment by tas50** [comment]
@sanga1794 There's a simple chefstyle violation that should get corrected to get the tests green as well.

**Comment by tas50** [comment]
Thanks @sanga1794

**Pull request merged**